### DISCUSSION TITLE
Be able to run tests without using Hardhat launcher

### DISCUSSION THREAD

**Discussion by jylin** [NomicFoundation/hardhat#4500] [Ideas]
It would be nice if all the hardhat-injected fixtures and libraries in test were available through normal imports. It would simplify test running if it didn't require hardhat as a special launcher. Would this be possible?

**Comment by fvictorio** [comment]
Can you give us more details on what you mean? Which things do you want available through normal imports?

**Reply by jylin** [reply]
Just run tests without using `hardhat test`(for example Lock.js). It magically injects ethers, but also some other things. It would be nice if all that were standard imports instead of injected by `hardhat test`.

**Reply by alcuadrado** [reply]
You can import everything from `"hardhat"` and use a different launcher/test runner.

**Reply by jylin** [reply]
It doesn't quite work. Gives this error:
```
HardhatError: HH9: Error while loading Hardhat's configuration.

You probably tried to import the "hardhat" module from your config or a file imported from it.
This is not possible, as Hardhat can't be initialized while its config is being defined.

To learn more about how to access the Hardhat Runtime Environment from different contexts go to https://hardhat.org/hre
```

**Reply by fvictorio** [reply]
If you are importing `"hardhat"` from a file that is directly or indirectly imported from the config, then you'll get that error. If you run the command with `---show-stack-traces`, the stack should show you which file is causing the issue.
